content/4. Deeltaken/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While.md

```diff
@@ -0,0 +1,355 @@
+while ()  
+{  
+    //Kies een nieuw random getal  
+    number = rand.Next(5, 8);  
+}  
+  
+// Print wanneer getal correct is.
+```
+
+> [!info]- Mogelijke uitwerking
+> ``` csharp runner
+> using System;
+> 
+>// Het juiste getal
+>int correctNumber = 7;
+>
+>// Kies een random getal.
+>Random rand = new Random();
+>int number = rand.Next(5, 8);
+>
+>// While-lus om te blijven raden
+>while (number != correctNumber)
+>{
+>	Console.WriteLine($"Nieuwe gok: {number}");
+>	Console.WriteLine("Helaas dat is niet correct.");
+>	number = rand.Next(5, 8);
+>}
+>
+>// Print wanneer getal correct is.
+>Console.WriteLine($"Nieuwe gok: {number}");
+>Console.WriteLine("Gefeliciteerd! Dat is correct.");
+>```
+
+---
+## Opdracht 2
+Zet de volgende casus om naar een While loop.
+
+### Specificaties
+* Maak een variabel `input` aan.
+* Controleer of de `input` een positief getal is en tel af naar 0 met behulp van een While-loop.
+* Toon in de console de getallen van het aftellen.
+
+### Verwachte output:
+Als input -1 is:
+```
+Voer een positief getal in:
+Ongeldig getal. Probeer opnieuw.
+Voer een positief getal in:
+```
+Als input 5 is:
+```
+Voer een positief getal in:
+5
+4
+3
+2
+1
+0
+```
+
+### Nu jij:
+``` csharp runner
+using System;
+
+// Kies een random getal. (tussen -5 & 15)
+Random rand = new Random();  
+int number = rand.Next(-5, 15);  
+  
+// Controleer of het getal positief is  
+while ()  
+{  
+    // Getal is ongeldig, kies een nieuw random getal.  
+    number = rand.Next(-5, 15);  
+}  
+  
+// Aftellen naar 0  
+while ()  
+{  
+}
+```
+
+> [!info]- Mogelijke uitwerking
+> ``` csharp
+> using System;
+> 
+>// Kies een random getal.
+>Random rand = new Random();
+>int number = rand.Next(-5, 15);
+>
+>// Controleer of het getal positief is
+>while (number < 0)
+>{
+>	Console.WriteLine($"Nieuwe gok is: {number}");
+>	Console.WriteLine("Ongeldig getal. Probeer opnieuw.");
+>	number = rand.Next(-2, 2);
+>}
+>        
+>Console.WriteLine($"Nieuwe gok is: {number}");
+ >       
+>// Aftellen naar 0
+>while (number >= 0)
+>{
+>	Console.WriteLine(number);
+>	number--;
+>}
+>```
+
+---
+## Opdracht 3
+Zet de volgende casus om naar een While loop.
+
+### Specificaties
+* Vraag de gebruiker om een reeks positieve getallen in te voeren.
+* Tel de ingevoerde getallen bij elkaar op.
+* Het programma stopt zodra de gebruiker "0" invoert.
+* Toon de som van alle ingevoerde getallen.
+
+### Verwachte output:
+Als invoer bijvoorbeeld `5, 3, 2, 0` is:
+```
+Voer een positief getal in (0 om te stoppen):  
+5  
+Voer een positief getal in (0 om te stoppen):  
+3  
+Voer een positief getal in (0 om te stoppen):  
+2  
+Voer een positief getal in (0 om te stoppen):  
+0  
+Het totaal van alle ingevoerde getallen is: 10  
+```
+
+### Nu jij:
+``` csharp runner
+using System;
+
+// Variabel om het totaal bij te houden  
+int count = 0;  
+  
+// Kies een random getal  
+Random rand = new Random();  
+int number = rand.Next(0, 20);  
```

Comment:
Dit is een te grote range, hier kan die loop heel lang mee bezig zijn, dus eerder 
```suggestion
int number = rand.Next(0, 7);  
```

Reply:
Hetzelfde geldt natuurlijk voor de rand.Next in de While-loop